src/model_repository_manager.cc

```diff
@@ -848,24 +959,54 @@ ModelRepositoryManager::GetModel(
   return status;
 }
 
+Status
+ModelRepositoryManager::FindModelIdentifier(const std::string& model_name, ModelIdentifier* model_id)
+{
+  auto git = global_map_.find(model_name);
+  if (git == global_map_.end()) {
+    return Status(Status::Code::INVALID_ARG, "Request for unknown model: '" + model_name + "' is not found");
+  }
+  switch (git->second.size())
+  {
+  case 0:
+    return Status(Status::Code::NOT_FOUND, "Identifier of model '" + model_name + "' is not found in global map");
+    break;
+  case 1: {
+    *model_id = *git->second.begin();
+    break;
+  }
+  default:
+    return Status(Status::Code::INVALID_ARG, "There are multiple identifiers of model '" + model_name + "' in global map, model namespace must be provided to resolve ambiguity.");
```

Comment:
Optional, but it could be good to include the number of identifiers (git->second.size()) in the error message.